turbopack/crates/turbopack-core/src/module_graph/mod.rs

```diff
@@ -653,6 +706,12 @@ impl ModuleGraph {
         Ok(idx)
     }
 
+    // Iterate over all nodes in the graph
+    // pub async fn iter_nodes(&self) -> Result<AllNodesIterator> {
+    // let graphs = self.get_graphs().await?;
+    // Ok(AllNodesIterator { inner: graphs })
+    // }
```

Comment:
ReadRef implements IntoIter

```suggestion
    // Iterate over all nodes in the graph
    pub async fn iter_nodes(&self) -> Result<impl Iterator<Item = ...>> {
       let graphs = self.get_graphs().await?;
       Ok(graphs.into_iter().flatten())
    }
```

Reply:
That would only work if `SingleModuleGraph` implemented `IntoIterator` as well.
And `.flat_map(|g| g.iter_nodes())` unfortunately also doesn't work because of `cannot return value referencing function parameter g`